skimage/filters/_blur_effect.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import scipy.ndimage as ndi
+
+from ..color import rgb2gray
+from . import sobel
+from ..util import img_as_float
+
+
+__all__ = ['blur_effect']
+
+
+def blur_effect(image, h_size=11, channel_axis=None):
+    """
+    Compute a metric that indicates the strength of blur in an image (0 for no
+    blur, 1 for maximal blur).
+
+    Parameters
+    ----------
+    image : ndarray
+        RGB or grayscale nD image. The input image is converted to grayscale
+        before computing the blur metric.
+    h_size : int, optional
+        Size of the re-blurring filter. Default is 11.
+    channel_axis : int or None, optional
+        If None, the image is assumed to be grayscale (single-channel).
+        Otherwise, this parameter indicates which axis of the array
+        corresponds to color channels.
+
+    Returns
+    -------
+
+    blur : float (0 to 1)
+        Blur metric: the maximum of blur metrics along all axes.
+    blur_table : list of floats
+        The i-th element is the blur metric along the i-th axis.
+
+    Notes
+    -----
+
+    `h_size` must keep the same value in order to compare results between
+    images. Most of the time, the default size (11) is enough. This means that
+    the metric can clearly discriminate blur up to an average 11x11 filter; if
+    blur is higher, the metric still gives good results but its values tend
+    towards an asymptote.
+
+    References
+    ----------
+    .. [1] Frederique Crete, Thierry Dolmiere, Patricia Ladret, and Marina
+       Nicolas "The blur effect: perception and estimation with a new
+       no-reference perceptual blur metric" Proc. SPIE 6492, Human Vision and
+       Electronic Imaging XII, 64920I (2007)
+       https://hal.archives-ouvertes.fr/hal-00232709
+       :DOI:`10.1117/12.702790`
+    """
+    # number of non-channel axes
+    n_axes = image.ndim
+
+    if channel_axis is not None:
+        if not isinstance(channel_axis, int) or channel_axis >= image.ndim:
```

Comment:
This should also error on `channel_axis < -image.ndim`, but I'm not sure we need to check these at all or just let `np.moveaxis` handle it? It seems `moveaxis` would return a `TypeError` for non-integer `channel_axis` and a `numpy.AxisError` if the axis is out of range.

The main benefit of us raising the error is that we can make a more user-friendly error message that mentions `channel_axis` by name instead of, for example:
```Python
np.moveaxis(np.random.randn(16, 32), 1.5, -1)
Traceback (most recent call last):

  File "<ipython-input-53-0100d5f06b0f>", line 1, in <module>
    np.moveaxis(np.random.randn(16, 32), 1.5, -1)

  File "<__array_function__ internals>", line 5, in moveaxis

  File "/home/lee8rx/mambaforge/envs/skimage_dev/lib/python3.9/site-packages/numpy/core/numeric.py", line 1461, in moveaxis
    source = normalize_axis_tuple(source, a.ndim, 'source')

  File "/home/lee8rx/mambaforge/envs/skimage_dev/lib/python3.9/site-packages/numpy/core/numeric.py", line 1391, in normalize_axis_tuple
    axis = tuple([normalize_axis_index(ax, ndim, argname) for ax in axis])

TypeError: 'float' object is not iterable
```

Reply:
Good point! I've decided to combine both, so the error starts with a message that mentions `channel_axis`: 752f5d7448f89b143214c9556141cfca26a61497 What do you think?